provider/test_helpers.go

```diff
@@ -0,0 +1,44 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/firehydrant/terraform-provider-firehydrant/firehydrant"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// testAccCheckRoleResourceDestroy verifies that role resources are properly cleaned up
+func testAccCheckRoleResourceDestroy() resource.TestCheckFunc {
```

Comment:
Nice.  Thanks for adding this.  We should see if there's an abstractable pattern for these more generally in the future, but this a great check for now.